drivers/sensor/nxp/mcux_lpcmp/mcux_lpcmp.c

```diff
@@ -0,0 +1,452 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_lpcmp
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <fsl_lpcmp.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
+#include <zephyr/drivers/sensor/mcux_lpcmp.h>
+#include <fsl_lpcmp.h>
+
+LOG_MODULE_REGISTER(mcux_lpcmp, CONFIG_SENSOR_LOG_LEVEL);
+
+struct mcux_lpcmp_config {
+	LPCMP_Type *base;
+	const struct pinctrl_dev_config *pincfg;
+#ifdef CONFIG_MCUX_LPCMP_TRIGGER
+	void (*irq_config_func)(const struct device *dev);
+#endif /* CONFIG_MCUX_LPCMP_TRIGGER */
+	bool output_enable: 1;
+	bool unfiltered: 1;
+	bool output_invert: 1;
+	lpcmp_hysteresis_mode_t hysteresis_level;
+	lpcmp_power_mode_t power_level;
+	lpcmp_functional_source_clock_t function_clock;
+};
+
+struct mcux_lpcmp_data {
+	lpcmp_config_t lpcmp_config;
+	lpcmp_dac_config_t dac_config;
+	lpcmp_filter_config_t filter_config;
+	lpcmp_window_control_config_t window_config;
+#ifdef CONFIG_MCUX_LPCMP_TRIGGER
+	const struct device *dev;
+	struct sensor_trigger rising_trigger;
+	sensor_trigger_handler_t rising_handler;
+	struct sensor_trigger falling_trigger;
+	sensor_trigger_handler_t falling_handler;
```

Comment:
The triggers need to be pointers. See #55130

Reply:
updated, please help to review, thanks.